nix/default.nix

```diff
@@ -1,12 +1,130 @@
 let
   sources = import ./sources.nix;
+
   pkgs = import sources.nixpkgs {
     config.allowUnfree = true;
     overlays = [
       # All wire-server specific packages
       (import ./overlay.nix)
-
     ];
   };
+
+  c-lib-out-deps = [
+    pkgs.cryptobox
+    pkgs.icu.out
+    pkgs.libsodium.out
+    pkgs.libxml2.out
+    pkgs.ncurses.out
+    pkgs.openssl.out
+    pkgs.pcre.out
+    pkgs.snappy.out
+    pkgs.zlib.out
+    pkgs.lzma.out
+  ];
+
+  compile-deps = pkgs.buildEnv {
+    name = "wire-server-compile-deps";
+    paths = [
+      pkgs.bash
+      pkgs.coreutils
+      pkgs.gnused
+      pkgs.gnugrep
+      pkgs.pkgconfig
+      pkgs.gawk
+      pkgs.git
+
+      pkgs.haskell.compiler.ghc8107
+      pkgs.protobuf
+
+      pkgs.cryptobox
+      pkgs.icu.dev
+      pkgs.libsodium.dev
+      pkgs.libxml2.dev
+      pkgs.ncurses.dev
+      pkgs.openssl.dev
+      pkgs.pcre.dev
+      pkgs.snappy.dev
+      pkgs.zlib.dev
+      pkgs.lzma.dev
+    ] ++ c-lib-out-deps;
+  };
+
+  # This performs roughly the same setup as direnv's load_prefix function, but
+  # only when invoking cabal. This means that we can set LD_LIBRARY_PATH just
+  # for cabal, as setting it in direnv can interfere with programs in the host
+  # system, especially for non-NixOS users.
+  cabal-wrapper = pkgs.writeShellScriptBin "cabal" ''
+    export CPATH="${compile-deps}/include"
+    export LD_LIBRARY_PATH="${compile-deps}/lib"
+    export LIBRARY_PATH="${compile-deps}/lib"
+    export PKG_CONFIG_PATH="${compile-deps}/lib/pkgconfig"
+    export PATH="${compile-deps}/bin"
+    export CONFIG_SHELL="${compile-deps}/bin/sh"
+    exec "${pkgs.cabal-install}/bin/cabal" "$@"
+  '';
+
+  # stack-deps.nix sets LD_LIBRARY_PATH, which could be incompatible with the
+  # system bash. To ensure that nix-shell invoked by stack uses the correct
+  # shell to build we set NIX_BUILD_SHELL here.
+  stack-wrapper = pkgs.writeShellScriptBin "stack" ''
+    export NIX_BUILD_SHELL="${pkgs.bash}/bin/bash"
+    exec "${pkgs.stack}/bin/stack" "$@"
+  '';
+
+  devPackages = [
+    (pkgs.haskell-language-server.override { supportedGhcVersions = [ "8107" ]; })
+    pkgs.cfssl
+    pkgs.crypto_cli
+    pkgs.ghcid
+    pkgs.gnumake
+    pkgs.gnused
+    pkgs.helm
+    pkgs.helmfile
+    pkgs.jq
+    pkgs.kind
+    pkgs.kubectl
+    pkgs.netcat
+    pkgs.niv
+    pkgs.ormolu
+    pkgs.python3
+    pkgs.rsync
+    pkgs.shellcheck
+    pkgs.wget
+    pkgs.yq
```

Comment:
Is this missing ghcid and hlint? :)

Reply:
`ghcid` is in line 78 (it got sorted alphabetically), and `hlint` wasn't in `dev-packages.nix` before either.